framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -331,19 +332,48 @@ public boolean implementsWidening() {
         public AnnotationMirror widenUpperBound(AnnotationMirror a1, AnnotationMirror a2) {
             AnnotationMirror lub = leastUpperBound(a1, a2);
             if (AnnotationUtils.areSameByClass(lub, IntRange.class)) {
-                Range range = getRange(lub);
-                if (range.isWithin(Byte.MIN_VALUE, Byte.MAX_VALUE)) {
+                Range lubRange = getRange(lub);
+                if (Range.IGNORE_OVERFLOW) {
+                    Range range1 = getRangeOrConvertIntVal(a1);
```

Comment:
@smillst I don't entirely understand why you added this code, either. Can you add the comment Mike requested?